Jenkinsfile

```diff
@@ -0,0 +1,244 @@
+/*
+Jenkins Pipeline is a suite of plugins which supports implementing and integrating continuous delivery pipelines into Jenkins.
+Pipeline provides an extensible set of tools for modeling delivery pipelines "as code" via the Pipeline DSL.
+More information can be found on the Jenkins Documentation page https://jenkins.io/doc/
+*/
+
+library 'github-utils-shared-library@master'
+@Library('github.com/connexta/cx-pipeline-library@master') _
+
+pipeline {
+    agent {
+        node {
+            label 'linux-small'
+            customWorkspace "/jenkins/workspace/${env.JOB_NAME}/${env.BUILD_NUMBER}"
+        }
+    }
+    environment {
+        DISABLE_DOWNLOAD_PROGRESS_OPTS = '-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn'
+        LINUX_MVN_RANDOM = '-Djava.security.egd=file:/dev/./urandom'
+        COVERAGE_EXCLUSIONS = '**/test/**/*,**/itests/**/*,**/*Test*,**/*.txt,**/*.xml'
+        GITHUB_USERNAME = 'connexta'
+        GITHUB_REPONAME = 'ion-transformation-service-registry'
+        GITHUB_KEY = 'ion-transformation-service-registry-github-key'
+    }
+    parameters {
+        booleanParam(name: 'RELEASE', defaultValue: false, description: 'Perform Release?')
+        string(name: 'RELEASE_VERSION', defaultValue: 'NA', description: 'The version to release. An NA value will release the current version')
+        string(name: 'RELEASE_TAG', defaultValue: 'NA', description: 'The release tag for this version. An NA value will result in <repository-name>-RELEASE_VERSION')
+        string(name: 'NEXT_VERSION', defaultValue: 'NA', description: 'The next development version. An NA value will increment the patch version')
+    }
+    options {
+        buildDiscarder(logRotator(numToKeepStr:'25'))
+        disableConcurrentBuilds()
+        timestamps()
+    }
+    triggers {
+        /*
+          Restrict nightly builds to master branch, all others will be built on change only.
+          Note: The BRANCH_NAME will only work with a multi-branch job using the github-branch-source
+        */
+        cron(env.BRANCH_NAME == "master" ? "H H(21-23) * * *" : "")
+    }
+    stages {
+        stage('Setup') {
+            steps {
```

Comment:
You need to call `dockerd{}` as the first step here if you're building dockerfiles.